pkg/k8s/apis/cilium.io/v2/register.go

```diff
@@ -151,5 +177,732 @@ func CreateCustomResourceDefinitions(clientset apiextensionsclient.Interface) er
+				"rules": {
+					Description: "Rules is a list of additional port level rules which must be " +
+						"met in order for the PortRule to allow the traffic. If omitted or empty, " +
+						"no layer 7 rules are enforced.",
+					Ref: getStr("#/properties/L7Rules"),
+				},
+			},
+		},
+		"PortRuleHTTP": {
+			Description: "PortRuleHTTP is a list of HTTP protocol constraints. All fields are " +
+				"optional, if all fields are empty or missing, the rule does not have any effect." +
+				"\n\nAll fields of this type are extended POSIX regex as defined by " +
+				"IEEE Std 1003.1, (i.e this follows the egrep/unix syntax, not the perl syntax) " +
+				"matched against the path of an incoming request. Currently it can contain " +
+				"characters disallowed from the conventional \"path\" part of a URL as defined by " +
+				"RFC 3986.",
+			Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+				"headers": {
+					Description: "Headers is a list of HTTP headers which must be present in the " +
+						"request. If omitted or empty, requests are allowed regardless of headers " +
+						"present.",
+					Type: "array",
+					Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
+						Schema: &apiextensionsv1beta1.JSONSchemaProps{
+							Type: "string",
+						},
+					},
+				},
+				"host": {
+					Description: "Host is an extended POSIX regex matched against the host header " +
+						"of a request, e.g. \"foo.com\"\n\nIf omitted or empty, the value of the " +
+						"host header is ignored.",
+					Type:   "string",
+					Format: "idn-hostname",
+				},
+				"method": {
+					Description: "Method is an extended POSIX regex matched against the method of " +
+						"a request, e.g. \"GET\", \"POST\", \"PUT\", \"PATCH\", \"DELETE\", ...\n\n" +
+						"If omitted or empty, all methods are allowed.",
+					Type: "string",
+					Enum: []apiextensionsv1beta1.JSON{
+						{
+							Raw: []byte(`"GET"`),
+						},
+						{
+							Raw: []byte(`"HEAD"`),
+						},
+						{
+							Raw: []byte(`"POST"`),
+						},
+						{
+							Raw: []byte(`"PUT"`),
+						},
+						{
+							Raw: []byte(`"DELETE"`),
+						},
+						{
+							Raw: []byte(`"CONNECT"`),
+						},
+						{
+							Raw: []byte(`"OPTIONS"`),
+						},
+						{
+							Raw: []byte(`"PATCH"`),
+						},
+					},
+				},
+				"path": {
+					Description: "Path is an extended POSIX regex matched against the path of a " +
+						"request. Currently it can contain characters disallowed from the " +
+						"conventional \"path\" part of a URL as defined by RFC 3986. Paths must " +
+						"begin with a '/'.\n\nIf omitted or empty, all paths are all allowed.",
+					Type:    "string",
+					Pattern: `^\/.*`,
+				},
+			},
+		},
+		"PortRuleKafka": {
+			Description: "PortRuleKafka is a list of Kafka protocol constraints. All fields are " +
+				"optional, if all fields are empty or missing, the rule will match all Kafka " +
+				"messages.",
+			Required: []string{
+				"apiKeyInt",
```

Comment:
ApiKey is the required field not ApiKetInt. ApikeyInt is used more as a "string" to "int" cache of the Apikey

Reply:
they are indeed unexported on the `PortRuleKafka` structure, I'll change this. However, the `APIKey` has a comment describing that it is optional:
```
	// If omitted or empty, all keys are allowed.
	//
	// +optional
	APIKey string `json:"apiKey,omitempty"`
```

Reply:
My Bad. I meant APIKeyInt is not mandatory. It is just a cached version of APIKey "string" to Int conversion and is only used internally by cilium. It is not an input in the policy (infact these fields APIKeyInt and APIVersionInt are private ). The APIKey is indeed a `optional` field as the comment mentions